src/boot/efi/meson.build

```diff
@@ -258,6 +258,9 @@ if have_gnu_efi
                        '-shared',
                        '-Bsymbolic',
                        '-nostdlib',
+                       '--no-undefined',
+                       '--warn-common',
+                       '--fatal-warnings',
```

Comment:
hmm, shouldn't we set that for all linker stuff?

Reply:
The sd-boot compile/link steps are hand crafted `custom_target`s.